### PR TITLE
use  in(char, string) rather than occursin(string,string)

### PR DESCRIPTION
**Pull request by oxinabox**
one of the things from
https://github.com/JuliaTime/TimeZones.jl/issues/282

### CODE REVIEW

File: src/parse.jl [hunk 1]
```diff
@@ -57,7 +57,7 @@ function tryparsenext_tz(str, i, len, min_width::Int=1, max_width::Int=0)
         # If the time zone is recognized make sure that it is well-defined. For our
         # purposes we'll treat all abbreviations except for UTC and GMT as ambiguous.
         # e.g. "MST": "Mountain Standard Time" (UTC-7) or "Moscow Summer Time" (UTC+3:31).
-        if occursin("/", name) || name in ("UTC", "GMT")
+        if in('/', name) || name in ("UTC", "GMT")
```
Comment (omus) [inline src/parse.jl:60]: May as well be consistent with the next condition
```suggestion
        if '/' in name || name in ("UTC", "GMT")
```

**Review by omus — COMMENTED** [review]
Would be good to get before/after benchmarks posted

**Comment by oxinabox** [comment]
compare https://github.com/JuliaTime/TimeZones.jl/issues/282#issuecomment-687169225
to

this PR:
```julia
julia> @btime parse(ZonedDateTime, "2017-11-14 11:03:53 +0100", dateformat"yyyy-mm-dd HH:MM:SS zzzzz");
  744.915 ns (13 allocations: 592 bytes)

julia> @btime parse(ZonedDateTime, "2016-04-11 08:00 UTC", dateformat"yyyy-mm-dd HH:MM ZZZ");
  660.746 ns (15 allocations: 704 bytes)

julia> @btime parse(ZonedDateTime, "2000+00", dateformat"yyyyz");
  464.526 ns (13 allocations: 496 bytes)

julia> @btime parse(ZonedDateTime, Test.GenericString("2018-01-01 00:00 UTC"), dateformat"yyyy-mm-dd HH:MM ZZZ");
  9.108 μs (136 allocations: 4.44 KiB)

julia> @btime tryparse(ZonedDateTime, "2013-03-20 11:00:00+04:00", dateformat"y-m-d H:M:SSz");
  888.667 ns (14 allocations: 640 bytes)

julia> @btime tryparse(ZonedDateTime, "2016-04-11 08:00 EST", dateformat"yyyy-mm-dd HH:MM zzz");
  90.784 ns (0 allocations: 0 bytes)

julia> @btime ZonedDateTime("2000-01-02T03:04:05.006+0700");
  830.208 ns (14 allocations: 688 bytes)

julia> @btime ZonedDateTime("2000-01-02T03:04:05.006Z");
  629.645 ns (9 allocations: 416 bytes)

julia> @btime ZonedDateTime("2018-11-01-0600", dateformat"yyyy-mm-ddzzzz");
  612.932 ns (13 allocations: 544 bytes)
```

The important on is:
```
julia> @btime parse(ZonedDateTime, Test.GenericString("2018-01-01 00:00 UTC"), dateformat"yyyy-mm-dd HH:MM ZZZ");
  9.108 μs (136 allocations: 4.44 KiB)
```
vs master:
```julia
julia> @btime parse(ZonedDateTime, Test.GenericString("2018-01-01 00:00 UTC"), dateformat"yyyy-mm-dd HH:MM ZZZ");
  9.600 μs (138 allocations: 4.48 KiB)
```

**Comment by omus** [comment]
I'll just note the first test here is slower than what occurred on `master` by 39 ns. Probably just noise but I wanted to point out that it was noticed.

**Comment by oxinabox** [comment]
😂  🦅 👁️ 
I also noticed but figured it was noise

**Comment by omus** [comment]
Looks like mainly we're seeing the performance bump from using a `Char` over a `String`:
```julia
julia> using Random, BenchmarkTools

julia> randstring(501) * '/' * randstring(498);

julia> str = randstring(501) * '/' * randstring(498);

julia> length(str)
1000

julia> @btime occursin("/", $str)
  30.878 ns (0 allocations: 0 bytes)
true

julia> @btime in('/', $str)
  20.098 ns (0 allocations: 0 bytes)
true

julia> @btime occursin('/', $str)
  21.028 ns (0 allocations: 0 bytes)
true
```
I'm still good with this.

**Pull request merged**